server/src/main/java/org/elasticsearch/search/aggregations/BucketOrder.java

```diff
@@ -129,7 +130,10 @@ public final void validate(Aggregator aggregator) throws AggregationExecutionExc
     /**
      * Build a comparator for {@link DelayedBucket}, a wrapper that delays bucket reduction.
      */
```

Comment:
Probably worth some javadoc warning that the context has to be the one for final reduction. It's pretty obvious to you and me, but I'm thinking of me-in-five-years who will have forgotten lots of things.

Reply:
added